src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -485,18 +372,19 @@ void Chttp2ServerListener::ActiveConnection::HandshakingState::OnHandshakeDone(
           GRPC_CLOSURE_INIT(&on_receive_settings_, OnReceiveSettings, this,
                             grpc_schedule_on_exec_ctx);
           // If the listener has been configured with a config fetcher, we
-          // need to watch on the transport being closed so that we can an
-          // updated list of active connections.
+          // need to watch on the transport being closed so that we can update
+          // the list of active connections.
           grpc_closure* on_close = nullptr;
-          if (connection_->listener_->config_fetcher_watcher_ != nullptr) {
+          if (connection_->listener()->server()->config_fetcher() != nullptr) {
```

Comment:
I think one of the goals here should be that the transport-specific code doesn't know anything about the config fetcher specifically.  And it looks like we're getting a callback when the transport is closed in either case.  Can we just unconditionally use the same callback in both cases, and have that callback call into the server code, which can remove the connection if it's tracking connections?

Reply:
I think we can simplify the code without giving up this optimization.  The server code already knows whether there is a config fetcher, so I think we can just make it such that both `ListenerWrapper::AddLogicalConnection()` and `ListenerWrapper::RemoveLogicalConnection()` are no-ops if there's no config fetcher.  Then the transport code can just unconditionally call those methods at the right points, and the right thing will happen.

(Now that you've split the new code into `NewChttp2ServerListener`, this comment applies to that new class, not to the old one.)

Reply:
This is the same `connection_quota` issue that we are talking about elsewhere. The difference between the with config fetcher and no config fetcher is -

With Config Fetcher -
Connection is tracked by Listener. When connection is closed, we reduce the connection count in the connection quota and remove the connection from the listener tracking.

Without Config Fetcher -
We remove the connection from the listener tracking when handshake is done but we still want to track the connection quota (which is want `OnClose` for the case without config fetcher is doing here).